specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

```diff
@@ -1,132 +1,224 @@
+            "in": "header",
+            "description": "serialization type for schema being registered",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Location": {
+                "type": "string"
+              },
+              "X-Serialization": {
+                "type": "string",
+                "description": "schema type"
+              },
+              "X-Schema-Id": {
+                "type": "string",
+                "description": "unique schema identifier"
+              },
+              "X-Schema-Id-Location": {
+                "type": "string",
+                "description": "location of schema resource"
+              },
+              "X-Schema-Version": {
+                "type": "integer",
+                "description": "version of returned schema"
+              }
+            },
             "schema": {
-              "$ref": "#/definitions/ErrorResponse"
+              "$ref": "#/definitions/SchemaId"
             }
           }
         }
       }
     }
   },
   "definitions": {
-    "Result": {
-      "description": "Sample result definition",
+    "SchemaId": {
       "properties": {
-        "sampleProperty": {
-          "type": "string",
-          "description": "Sample property of type string"
+        "id": {
+          "type": "string"
         }
-      }
-    },
-    "ErrorResponse": {
-      "description": "Error response.",
-      "properties": {
-        "error": {
-          "$ref": "#/definitions/ErrorDefinition",
-          "description": "The error details."
-        }
-      }
-    },
-    "ErrorDefinition": {
-      "description": "Error definition.",
-      "properties": {
-        "code": {
-          "description": "Service specific error code which serves as the substatus for the HTTP error code.",
-          "type": "string",
-          "readOnly": true
-        },
-        "message": {
-          "description": "Description of the error.",
-          "type": "string",
-          "readOnly": true
-        },
-        "details": {
-          "description": "Internal error details.",
-          "type": "array",
-          "items": {
-            "$ref": "#/definitions/ErrorDefinition"
-          },
-          "readOnly": true
-        }
-      }
-    }
-  },
-  "parameters": {
-    "SubscriptionIdParameter": {
-      "name": "subscriptionId",
-      "in": "path",
-      "required": true,
-      "type": "string",
-      "description": "The Azure subscription ID. This is a GUID-formatted string (e.g. 00000000-0000-0000-0000-000000000000)"
-    },
-    "ResourceGroupNameParameter": {
-      "name": "resourceGroupName",
-      "in": "path",
-      "required": true,
-      "type": "string",
-      "description": "The name of the resource group.",
-      "x-ms-parameter-location": "method"
-    },
-    "ApiVersionParameter": {
```

Comment:
Do you not use an API version query parameter to differentiate endpoint versions?

Reply:
We do, but my understanding is that all Azure core libraries have header policies where the API version is specified when building the object.  At least that's how we did it in Java on the HttpPipeline object that's passed to the RestService builder.

Reply:
I suppose that the version can be set on the layer above the RestService object and can be passed in as a parameter to all RestService method calls.

Reply:
If you model it like [this](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/artifacts.json#L4313) the code generators will do the right thing.

Reply:
Added.